ocs_ci/ocs/perf_pgsql.py

```diff
@@ -0,0 +1,87 @@
+"""
+PGSQL performance workload class
+"""
+import logging
+
+from prettytable import PrettyTable
+
+from ocs_ci.ocs.pgsql import Postgresql
+
+log = logging.getLogger(__name__)
+
+
+class PerfPGSQL(Postgresql):
+    """
+    Class to create pgsql, pgbench pods
+    and run workloads
+
+    """
+
+    def __init__(self, **kwargs):
+        """
+        - Initializer function
+        - Deploys Benchmark Operator
+
+        """
+        super().__init__(**kwargs)
+        Postgresql.deploy(self)
+
+    def _setup_postgresql(self, replicas):
+        Postgresql.setup_postgresql(self, replicas=replicas)
+
+    def _create_pgbench_benchmark(
+        self,
+        replicas,
+        clients=None,
+        threads=None,
+        transactions=None,
+        scaling_factor=None,
+        timeout=None,
+    ):
+        Postgresql.create_pgbench_benchmark(
+            self,
+            replicas,
+            clients=clients,
+            threads=threads,
+            transactions=transactions,
+            scaling_factor=scaling_factor,
+            timeout=timeout,
+        )
+
+    def _wait_for_pgbench_status(self, status, timeout=None):
+        Postgresql.wait_for_postgres_status(self, status=status, timeout=timeout)
+
+    def _get_pgbench_pods(self):
+        Postgresql.get_pgbench_pods(self)
+
+    def validate_pgbench_perf(self, pgbench_pods):
+        pgbench_pods_output = Postgresql.validate_pgbench_run(
+            self, pgbench_pods, print_table=True
+        )
+        tps_latency_table = PrettyTable()
+        tps_latency_table.field_names = [
+            "pod_name",
+            "latency_avg",
+            "lat_stddev",
+            "tps_incl",
+            "tps_excl",
+        ]
+        # Taking transaction per second and latency information
+        for pgbench_pod_out in pgbench_pods_output:
+            for pod_output in pgbench_pod_out[0]:
+                for pod in pod_output.values():
+                    tps_latency_table.add_row(
```

Comment:
@keemano Looks like all the pod data will be one after another in table and hope `pgbench_pod_out[0]` 0th position won't be changed in future.

Reply:
> @keemano Looks like all the pod data will be added one after another in table and hope `pgbench_pod_out[0]` 0th position won't be changed in future.

@ramkiperiy Yes "pgbench_pod_out[0]" will always contain the output data like latency, tps, etc. 1st position contain the pod name. It will not change in the future.

FYI

"pgbench_pods_output" will  be in the below format

[([{1: {'scaling_factor': '2', 'num_clients': '20', 'num_threads': '2', 'number_of_transactions_per_client': '100', 'number_of_transactions_actually_processed': '2000', 'latency_avg': '70', 'lat_stddev': '59', 'tps_incl': '264', 'tps_excl': '264'}}, {2: {'scaling_factor': '2', 'num_clients': '20', 'num_threads': '2', 'number_of_transactions_per_client': '100', 'number_of_transactions_actually_processed': '2000', 'latency_avg': '71', 'lat_stddev': '61', 'tps_incl': '265', 'tps_excl': '265'}}, {3: {'scaling_factor': '2', 'num_clients': '20', 'num_threads': '2', 'number_of_transactions_per_client': '100', 'number_of_transactions_actually_processed': '2000', 'latency_avg': '69', 'lat_stddev': '59', 'tps_incl': '271', 'tps_excl': '271'}}, {4: {'scaling_factor': '2', 'num_clients': '20', 'num_threads': '2', 'number_of_transactions_per_client': '100', 'number_of_transactions_actually_processed': '2000', 'latency_avg': '73', 'lat_stddev': '71', 'tps_incl': '251', 'tps_excl': '252'}}], 'pgbench-1-dbs-client-1-e57e91c4--1-ftf9h')]